agent/app/agent_unix.go

```diff
@@ -0,0 +1,156 @@
+// +build linux
+
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package app
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/aws/amazon-ecs-agent/agent/ec2"
+	"github.com/aws/amazon-ecs-agent/agent/ecscni"
+	"github.com/aws/amazon-ecs-agent/agent/engine"
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate"
+	"github.com/aws/amazon-ecs-agent/agent/eni/pause"
+	enisetup "github.com/aws/amazon-ecs-agent/agent/eni/setup"
+)
+
+// initPID defines the process identifier for the init process
+const initPID = 1
+
+// awsVPCCNIPlugins is a list of CNI plugins required by the ECS Agent
+// to configure the ENI for a task
+var awsVPCCNIPlugins = []string{ecscni.ECSENIPluginName,
+	ecscni.ECSBridgePluginName,
+	ecscni.ECSIPAMPluginName,
+}
+
+// initializeTaskENIDependencies initializes all of the dependencies required by
+// the Agent to support the 'awsvpc' networking mode. A non nil error is returned
+// if an error is encountered during this process. An additional boolean flag to
+// indicate if this error is considered terminal is also returned
```

Comment:
I'm wondering if we should make the return something like `ok, err` rather than `err, notOK`.  I think having `true` mean that it's not okay is weird to me.  I can be convinced that what you have here is reasonable enough.